client/my-sites/checkout/checkout-thank-you/licensing-thank-you-auto-activation.tsx

```diff
@@ -81,6 +84,19 @@ const LicensingActivationThankYou: FC< Props > = ( {
 	const [ selectedSite, setSelectedSite ] = useState( '' );
 	const [ error, setError ] = useState< TranslateResult | false >( false );
 
+	const initialSelectedSite = useMemo( () => {
+		const validSiteThatMatchesFromSiteSlugProp = ( site: JetpackSite ) =>
+			site.is_wpcom_atomic === false && site.slug === fromSiteSlug;
+
+		return (
+			( !! fromSiteSlug && jetpackSites.find( validSiteThatMatchesFromSiteSlugProp )?.URL ) || ''
+		);
```

Comment:
This is already quite clear, but I'd say we could return early if there's no `fromSiteSlug`, that way we won't have to check it here (and avoid having an even bigger ternary in the future)